internal/services/cosmos/cosmosdb_cassandra_datacenter_resource.go

```diff
@@ -120,73 +120,68 @@ func resourceCassandraDatacenter() *pluginsdk.Resource {
 }
 
 func resourceCassandraDatacenterCreate(d *pluginsdk.ResourceData, meta interface{}) error {
-	client := meta.(*clients.Client).Cosmos.CassandraDatacentersClient
+	client := meta.(*clients.Client).Cosmos.ManagedCassandraClient
 	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	clusterId, _ := parse.CassandraClusterID(d.Get("cassandra_cluster_id").(string))
-	id := parse.NewCassandraDatacenterID(clusterId.SubscriptionId, clusterId.ResourceGroup, clusterId.Name, d.Get("name").(string))
+	clusterId, _ := managedcassandras.ParseCassandraClusterID(d.Get("cassandra_cluster_id").(string))
+	id := managedcassandras.NewDataCenterID(clusterId.SubscriptionId, clusterId.ResourceGroupName, clusterId.CassandraClusterName, d.Get("name").(string))
```

Comment:
missed this yesterday, apolgoies - but we also need to update line 37 to use this ID parser - and remove the old parser from `resourceids.go` (and the generated files from `parse` and `validate`) - can we apply that change too?

Reply:
Updated.